api/firmware/query.go

```diff
@@ -70,27 +71,37 @@ func (device *Device) rawQueryV7(msg []byte) ([]byte, error) {
 		break
 	}
 
-	var lastQueryTime time.Time
-
+	// saving 11 timestamps to show the last 10 intervals.
+	lastQueryTimes := ring.New(11)
 	for {
 		switch status {
 		case hwwRspAck:
 			return payload, nil
 		case hwwRspBusy:
 			return nil, errp.New("unexpected hwwRspBusy response")
 		case hwwRspNack:
-			if lastQueryTime.IsZero() {
+			lastQueryTimes = lastQueryTimes.Prev()
+			if lastQueryTimes.Prev().Value == nil {
 				device.log.Debug("unexpected NACK response in first loop iteration")
 			} else {
-				device.log.Debug(
-					fmt.Sprintf(
-						"unexpected NACK response; last successful query was %v ago",
-						time.Since(lastQueryTime)))
+				device.log.Debug("unexpected NACK response; last successful retry query intervals (newest on top) were:")
+				for i := 0; i < lastQueryTimes.Len()-1; i++ {
+					if lastQueryTimes.Value == nil || lastQueryTimes.Prev().Value == nil {
+						break
+					}
+					t := lastQueryTimes.Value
+					lastQueryTimes = lastQueryTimes.Prev()
+					tPrev := lastQueryTimes.Value
+					device.log.Debug(
```

Comment:
i think it would be easier to read the log if the whole string was one log-line instead of one log entry per interval.

Reply:
Good point, fixed